src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -427,5 +457,51 @@ private async Task AssertAutomaticReviewModifier(ClaimsPrincipal user, ReviewMod
                 throw new AuthorizationFailedException();
             }
         }
+
+        private async Task ApproveOrOutdateReviews(ClaimsPrincipal user, ReviewModel review, string revisionId)
```

Comment:
Why do we have logic in both directions? One that approves similar revisions and another to collect approvals? Seems that one should be enough.

Reply:
I have added checks in both directions to ensure all scenarios are covered. If manual review is already approved(if manual review was not created from master), then matching auto review may not be present at approval time.  So checking for a matching approved review at time automatic review creation finds a match and mark it approved.

If manual review is created after merging it on to master then automatic review may be already present. So checking for existing reviews for same package when one is approved ensures existing automatic review is matched and approved. This step also ensures of marking other pending reviews for same package as outdated so we can block approval of two different changes for same package at same time which takes care of two ppl working on same package in parallel and creates API reviews in parallel.

Reply:
Removed the logic when approve button is clicked.